lib/android.js

```diff
@@ -29,12 +29,22 @@ function getApi () {
       },
       '_ZN3art6Thread14CurrentFromGdbEv': ['art::Thread::CurrentFromGdb', 'pointer', []],
       '_ZNK3art6Thread13DecodeJObjectEP8_jobject': ['art::Thread::DecodeJObject', 'pointer', ['pointer', 'pointer']],
-      '_ZN3art6mirror6Object5CloneEPNS_6ThreadE': ['art::mirror::Object::Clone', 'pointer', ['pointer', 'pointer']]
     },
     variables: {
       '_ZN3art7Runtime9instance_E': function (address) {
         this.runtime_instance_ptr = address;
       }
+    },
+    extraInit: function(exports, fullAPI) {
+      fullAPI.isXposed = '_ZNK3art9OatHeader23IsXposedOatVersionValidEv' in exports;
+      let exp = exports[fullAPI.isXposed ? '_ZN3art6mirror6Object5CloneEPNS_6ThreadEm' :
+       '_ZN3art6mirror6Object5CloneEPNS_6ThreadE'];
+      if (exp !== undefined) {
+        fullAPI['art::mirror::Object::Clone'] = new NativeFunction(exp.address, 'pointer', 
+          fullAPI.isXposed ? ['pointer', 'pointer', 'int'] : ['pointer', 'pointer']);
+        return true;
+      }
+      return false;
     }
```

Comment:
I see. Currently `optionals` doesn't have support for `remaining` counter, so in case if none of these exports could be found it'll still return a non-functional API object. And extending `optionals` to group and count exports by their name will result in similar, if not more, complex code.